tsl/src/bgw_policy/policy_utils.c

```diff
@@ -64,43 +64,61 @@ policy_config_check_hypertable_lag_equality(Jsonb *config, const char *json_labe
 	}
 }
 
-Datum
-subtract_integer_from_now(int64 interval, Oid time_dim_type, Oid now_func)
+int64
+subtract_integer_from_now_internal(int64 interval, Oid time_dim_type, Oid now_func, bool *overflow)
 {
 	Datum now;
-	int64 res;
+	int64 res = 0;
 
 	AssertArg(IS_INTEGER_TYPE(time_dim_type));
 
 	now = OidFunctionCall0(now_func);
-
+	*overflow = false;
 	switch (time_dim_type)
 	{
 		case INT2OID:
 			res = DatumGetInt16(now) - interval;
 			if (res < PG_INT16_MIN || res > PG_INT16_MAX)
-				ereport(ERROR,
-						(errcode(ERRCODE_INTERVAL_FIELD_OVERFLOW), errmsg("ts_interval overflow")));
-			return Int16GetDatum(res);
+				*overflow = true;
+			break;
 		case INT4OID:
 			res = DatumGetInt32(now) - interval;
 			if (res < PG_INT32_MIN || res > PG_INT32_MAX)
-				ereport(ERROR,
-						(errcode(ERRCODE_INTERVAL_FIELD_OVERFLOW), errmsg("ts_interval overflow")));
-			return Int32GetDatum(res);
+				*overflow = true;
```

Comment:
Would be good to hit this in a test. Perhaps a C-based unit test is best to test this corner-cases?